src/pip/_internal/models/link.py

```diff
@@ -193,6 +193,12 @@ def is_artifact(self):
 
         return True
 
+    @property
+    def is_vcs(self):
+        from pip._internal.vcs import vcs
+
+        return self.scheme in vcs.all_schemes
```

Comment:
Since with this change this logic is getting duplicated in both `is_vcs` and `is_artifact`, I think it would be best if `is_artifact` is changed to call this method. (And you can move `is_vcs` before `is_artifact` so `is_artifact` is depending on methods defined before it.) Also, that way we avoid introducing a second nested import.

Reply:
Good point. Done.